moment.d.ts

```diff
@@ -0,0 +1,483 @@
+    * @param amount the amount you want to subtract
+    */
+    subtract(amount: number, unitOfTime: string): Moment;
+    /**
+    * Mutates the original moment by subtracting time. Note that the order of arguments can be flipped.
+    *
+    * @param amount the amount you want to add
+    * @param unitOfTime the unit of time you want to subtract (eg "years" / "hours" etc)
+    */
+    subtract(amount: string, unitOfTime: string): Moment;
+    /**
+    * Mutates the original moment by subtracting time.
+    *
+    * @param objectLiteral an object literal that describes multiple time units {days:7,months:1}
+    */
+    subtract(objectLiteral: MomentInput): Moment;
+    /**
+    * Mutates the original moment by subtracting time.
+    *
+    * @param duration a length of time
+    */
+    subtract(duration: Duration): Moment;
+
+    calendar(): string;
+    calendar(start: Moment): string;
+    calendar(start: Moment, formats: MomentCalendar): string;
+
+    clone(): Moment;
+
+    /**
+    * @return Unix timestamp, or milliseconds since the epoch.
+    */
+    valueOf(): number;
+
+    local(): Moment; // current date/time in local mode
+
+    utc(): Moment; // current date/time in UTC mode
+
+    isValid(): boolean;
+    invalidAt(): number;
+
+    year(y: number): Moment;
+    year(): number;
+    quarter(): number;
+    quarter(q: number): Moment;
+    month(M: number): Moment;
+    month(M: string): Moment;
+    month(): number;
+    day(d: number): Moment;
+    day(d: string): Moment;
+    day(): number;
+    date(d: number): Moment;
+    date(): number;
+    hour(h: number): Moment;
+    hour(): number;
+    hours(h: number): Moment;
+    hours(): number;
+    minute(m: number): Moment;
+    minute(): number;
+    minutes(m: number): Moment;
+    minutes(): number;
+    second(s: number): Moment;
+    second(): number;
+    seconds(s: number): Moment;
+    seconds(): number;
+    millisecond(ms: number): Moment;
+    millisecond(): number;
+    milliseconds(ms: number): Moment;
+    milliseconds(): number;
+    weekday(): number;
+    weekday(d: number): Moment;
+    isoWeekday(): number;
+    isoWeekday(d: number): Moment;
+    weekYear(): number;
+    weekYear(d: number): Moment;
+    isoWeekYear(): number;
+    isoWeekYear(d: number): Moment;
+    week(): number;
+    week(d: number): Moment;
+    weeks(): number;
+    weeks(d: number): Moment;
+    isoWeek(): number;
+    isoWeek(d: number): Moment;
+    isoWeeks(): number;
+    isoWeeks(d: number): Moment;
+    weeksInYear(): number;
+    isoWeeksInYear(): number;
+    dayOfYear(): number;
+    dayOfYear(d: number): Moment;
+
+    from(f: Moment | string | number | Date | number[], suffix?: boolean): string;
+    to(f: Moment | string | number | Date | number[], suffix?: boolean): string;
+    toNow(withoutPrefix?: boolean): string;
+
+    diff(b: Moment): number;
+    diff(b: Moment, unitOfTime: string): number;
+    diff(b: Moment, unitOfTime: string, round: boolean): number;
```

Comment:
Do you know if TypeScript uses the JSDoc comments today? The next question would be, could the TypeScript compiler generate the defaults JSDoc comments.